src/sirhelpers.c

```diff
@@ -30,7 +30,7 @@ bool __sir_validptrptr(const void* restrict* pp, bool fail) {
     bool valid = NULL != pp;
     if (!valid && fail) {
         (void)_sir_seterror(_SIR_E_NULLPTR);
-        SIR_ASSERT("!NULL pointer");
+        SIR_ASSERT(!(bool)"NULL pointer!");
```

Comment:
I don't think you need the (bool) at least I have never used it unless one of your psycho checks doesn't let it through

Reply:
which compiler warning flag caused this to be marked as a warning?

Reply:
n/m, it's string-conversion